craft_parts/executor/executor.py

```diff
@@ -99,6 +99,8 @@ def prologue(self) -> None:
                     self._project_info.overlay_mount_dir, self._project_info
                 )
                 ctx.refresh_packages_list()
+                logger.info("Upgrading packages in the overlay base")
+                ctx.upgrade_packages()
```

Comment:
@linostar have you tested this with Rockcraft? I'm don't remember the details but I'm almost sure that the contents of this `PackageCacheMount` layer do not end up in the final lifecycle payload. Typically we update the packages list and then mount [a layer on top of this one](https://github.com/canonical/craft-parts/blob/main/craft_parts/executor/part_handler.py#L240) to install overlay-packages and execute the overlay-script.